cmd/tuf/server/main.go

```diff
@@ -43,104 +43,122 @@ var (
+		meta, err := local.GetMeta()
+		if err != nil {
+			logging.FromContext(ctx).Panicf("Getting meta: %v", err)
+		}
+		rootJSON, ok := meta["root.json"]
+		if !ok {
+			logging.FromContext(ctx).Panicf("Getting root: %v", err)
+		}
 
-	nsSecret := clientset.CoreV1().Secrets(ns)
-	if err := secret.ReconcileSecret(ctx, *secretName, ns, data, nsSecret); err != nil {
-		logging.FromContext(ctx).Panicf("Failed to reconcile secret %s/%s: %v", ns, *secretName, err)
-	}
-	// Serve the TUF repository.
-	logging.FromContext(ctx).Infof("tuf repository was created in: %s", dir)
-	serveDir := filepath.Join(dir, "repository")
-	logging.FromContext(ctx).Infof("tuf repository was created in: %s serving tuf root at %s", dir, serveDir)
-	fs := http.FileServer(http.Dir(serveDir))
-	http.Handle("/", fs)
-
-	/* #nosec G114 */
-	if err := http.ListenAndServe(":8080", nil); err != nil {
-		panic(err)
+		data := make(map[string][]byte)
+		data["root"] = rootJSON
+
+		// Then compress the root directory and put it into a secret
+		// Secrets have 1MiB and the repository as tested goes to about ~3k, so no
+		// worries here.
+		var compressed bytes.Buffer
+		if err := repo.CompressFS(os.DirFS(dir), &compressed, map[string]bool{"keys": true, "staged": true}); err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to compress the repo: %v", err)
+		}
+		data["repository"] = compressed.Bytes()
+
+		nsSecret := clientset.CoreV1().Secrets(ns)
+		if err := secret.ReconcileSecret(ctx, *secretName, ns, data, nsSecret); err != nil {
+			logging.FromContext(ctx).Panicf("Failed to reconcile secret %s/%s: %v", ns, *secretName, err)
+		}
+		// Serve the TUF repository.
+		logging.FromContext(ctx).Infof("tuf repository was created in: %s", dir)
+		serveDir := filepath.Join(dir, "repository")
+		logging.FromContext(ctx).Infof("tuf repository was created in: %s serving tuf root at %s", dir, serveDir)
+		fs := http.FileServer(http.Dir(serveDir))
+		http.Handle("/", fs)
+
+		/* #nosec G114 */
+		if err := http.ListenAndServe(":8080", nil); err != nil {
+			panic(err)
+		}
+	} else if buildEnv == "container" {
+		tufFiles, err := os.ReadDir(*dir)
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("failed to read dir %s: %v", *dir, err)
+		}
+		trimDir := strings.TrimSuffix(*dir, "/")
+		files := map[string][]byte{}
+
+		files = tuf.ProcessTufFiles(ctx, tufFiles, trimDir)
+		local, dir, err := repo.CreateRepo(ctx, files)
+		if err != nil {
+			logging.FromContext(ctx).Panicf("Failed to create repo: %v", err)
+		}
+		meta, err := local.GetMeta()
+		if err != nil {
+			logging.FromContext(ctx).Panicf("Getting meta: %v", err)
+		}
+		rootJSON, ok := meta["root.json"]
+		if !ok {
+			logging.FromContext(ctx).Panicf("Getting root: %v", err)
+		}
+
+		data := make(map[string][]byte)
+		data["root"] = rootJSON
+
+		// Then compress the root directory and put it into a secret
+		// Secrets have 1MiB and the repository as tested goes to about ~3k, so no
+		// worries here.
```

Comment:
Really minor nit, but sometimes comments are important for folks coming along behind. This is copy paste from the kube implementation above. Maybe it could be modified to indicate that it's simply compressing the root directory and will be served from the filesystem.